src/utils/batchRunAll.ts

```diff
@@ -0,0 +1,39 @@
+import * as CliProgress from 'cli-progress';
+
+export async function batchRunAll<I>(
+    maxBatchSize: number,
+    inputs: I[],
+    cb: (input: I) => Promise<void>,
+    progress: boolean
+): Promise<void> {
+    const i = [...inputs];
+    let completedJobs = 0;
+    const initialWorkingSet = i.splice(0, maxBatchSize);
+
+    let progressBar: CliProgress.SingleBar | undefined;
+    if (progress) {
+        progressBar = new CliProgress.SingleBar(
+            {
+                etaBuffer: 20,

```

Comment:
since resolution tends to happen in chunks, this should probably just be maxBatchSize. The ETA buffer is just a list of numbers, so keeping a big buffer is cheap.